### PR TITLE
UnitTesting Controllers

### PR DESCRIPTION
**Pull request by luci9876**

### CODE REVIEW

File: API/HrApi/Controllers/EmployeesController.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@ public EmployeesController(HrContext context)
             _context = context;
         }
 
-        
+
         [HttpGet]
```
Comment (IRailean) [inline API/HrApi/Controllers/EmployeesController.cs:22]: Why do you need EmployeeController.cs?

In your app can an Employee be present in DB without being assigned to a company?

File: API/CompanyTest.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        }
+
+        [Fact]
+        public async Task EditCompany_ReturnsNoContent_WhenCompanyIsUpdated()
+        {
+            var controller = new CompaniesController(_context);
+
+            var result = await controller.PutCompany(1, new Company { Id = 1, Name = "company" });
+
+            Assert.IsType<OkObjectResult>(result);
```
Comment (IRailean) [inline API/CompanyTest.cs:120]: You're updating the company with a new name. Consider adding another assert where you verify if a new name was set.

File: API/CompanyTest.cs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        {
+            var controller = new CompaniesController(_context);
+
+            var result = await controller.PutCompany(99, new Company { Id = 99, Name = "company", Description = "description" });
+
+            Assert.IsType<NotFoundResult>(result);
+        }
+
+        [Fact]
+        public async Task EditCompany_ReturnsNoContent_WhenCompanyIsUpdated()
```
Comment (IRailean) [inline API/CompanyTest.cs:114]: Your tests are highly coupled.

After EditCompany_ReturnsNoContent_WhenCompanyIsUpdated, your database will be changed, and other tests may be influenced by that.

You may want to create a new in-memory DB for each specific test to isolate them from one another.

File: API/HrApi/Controllers/EmployeesController.cs (hunk 1 above)
Comment (luci9876) [reply API/HrApi/Controllers/EmployeesController.cs:22]: Yes, I think that would be better-called PossibleEmployees

**Pull request merged**